### DISCUSSION TITLE
Unable to flash tasmota on Sonoff Dual R3 v2

### DISCUSSION THREAD

**Discussion by uzunix** [arendst/Tasmota#20249] [Q&A] [answered]
Hi Guys!
I am trying to flash tasmota on Sonoff Dual R3 v2. I am using the latest release (13.3.0.1), with binary tasmota32.factory.bin, but I am getting very strange behavior:

```bash
esptool.py --chip esp32 --port /dev/ttyUSB0 --baud 460800 write_flash 0x00000 tasmota32.factory.bin 
esptool.py v4.6.2
Serial port /dev/ttyUSB0
Connecting...
Chip is ESP32-D0WD-V3 (revision v3.0)
Features: WiFi, BT, Dual Core, 240MHz, VRef calibration in efuse, Coding Scheme None
Crystal is 40MHz
MAC: c8:f0:9e:a5:c6:4d
Stub is already running. No upload is necessary.
Changing baud rate to 460800
Changed.
Configuring flash size...
Flash will be erased from 0x00000000 to 0x002bafff...
Compressed 2861376 bytes to 1795084...
Writing at 0x00011bff... (1 %)Traceback (most recent call last):
  File "/home/user/.local/lib/python3.10/site-packages/esptool/__init__.py", line 1073, in _main
    main()
  File "/home/user/.local/lib/python3.10/site-packages/esptool/__init__.py", line 868, in main
    operation_func(esp, args)
  File "/home/user/.local/lib/python3.10/site-packages/esptool/cmds.py", line 595, in write_flash
    esp.flash_defl_block(block, seq, timeout=timeout)
  File "/home/user/.local/lib/python3.10/site-packages/esptool/loader.py", line 131, in inner
    return func(*args, **kwargs)
  File "/home/user/.local/lib/python3.10/site-packages/esptool/loader.py", line 1037, in flash_defl_block
    self.check_command(
  File "/home/user/.local/lib/python3.10/site-packages/esptool/loader.py", line 436, in check_command
    val, data = self.command(op, data, chk, timeout=timeout)
  File "/home/user/.local/lib/python3.10/site-packages/esptool/loader.py", line 405, in command
    p = self.read()
  File "/home/user/.local/lib/python3.10/site-packages/esptool/loader.py", line 338, in read
    return next(self._slip_reader)
StopIteration

A fatal error occurred: The chip stopped responding.
```

Here is some more output:
```bash
esptool.py chip_id
esptool.py v4.6.2
Found 2 serial ports
Serial port /dev/ttyUSB0
Connecting.....
Detecting chip type... Unsupported detection protocol, switching and trying again...
Connecting...
Detecting chip type... ESP32
Chip is ESP32-D0WD-V3 (revision v3.0)
Features: WiFi, BT, Dual Core, 240MHz, VRef calibration in efuse, Coding Scheme None
Crystal is 40MHz
MAC: c8:f0:9e:a5:c6:4d
Uploading stub...
Running stub...
Stub running...
Warning: ESP32 has no Chip ID. Reading MAC instead.
MAC: c8:f0:9e:a5:c6:4d
Hard resetting via RTS pin...

```
After this I can still reboot the device and Sonoff's original firmware loads...

Any help will be appreciated :)

**Comment by Jason2866** [comment]
You can try to reduce flash speed to 115200

**Reply by uzunix** [reply]
No, got the same error: A fatal error occurred: The chip stopped responding.

**Reply by sfromis** [reply, marked as answer]
I'd check two things. First if your USB/Serial adapter and power source is good for flashing. Many adapters can't deliver enough current, or you may not have enough available mA on your PC USB port. And make sure that soldering and connections are good.

**Reply by uzunix** [reply]
I think you're right. I was able to get to 20% writing just by changing the usb port. I think my adapter is the cause, probably can't provide more current. I have to think how to address this: using external power source or changing the adapter...

**Reply by joba-1** [reply]
adding a half full lipo or 2 1.5V batteries does it for me

**Reply by uzunix** [reply]
@joba-1 Batteries did not work for me - got the same error.
I end up using external power source on 3.3 volts, that got it. Thanks @sfromis for pointing me to the right direction.
Thanks everybody for participating in my problem.
I hope this discussion could save some time and effort on someone else.

**Reply by joba-1** [reply]
strange, normal aa batteries can deliver more than these devices can chew. If not near empty of course :). And lipos even more. Must be a cabling / connection problem.

**Reply by uzunix** [reply]
@joba-1 Two brand new AA batteries can get up to 3.1 ~ 3.15 volts, they could never reach 3.3 volts, which the chip requires. It's not just the mA, the volts matter too I guess...
I never tried with lipos, as I don't have anything on 3.3 lying around...

**Reply by sfromis** [reply]
While the nominal voltage is indeed 3.3V, the data sheet says 2.3V~3.6V

**Reply by joba-1** [reply]
exactly, and 500mA (for the chip alone though, not counting external cirquitry). Don't use button cells, anything else should work - and has always worked for me.

**Reply by uzunix** [reply]
Guys, I am really grateful for the help, but can you assume for the moment, that perhaps there is something around the chip circuitry, which prevents operating on lower voltages or perhaps the manufacturer changed something, because they have different components available..? There are so many reasons to encounter this behavior... Don't get me wrong, I am not saying that putting it on batteries is not working, I am just saying that it did not worked for me, with this particular unit...
Anyway in time, I am planning to buy more units of this model and if you're interested how the flashing process went with the other units, I can certainly share my experience...
